src/index/TextScoring.h

```diff
@@ -69,7 +70,11 @@ class ScoreData {
       const Index::TextVocab& textVocab, size_t& wordNotFoundErrorMsgCount);
 
   void calculateAVDL() {
-    averageDocumentLength_ =
-        nofDocuments_ ? (totalDocumentLength_ / nofDocuments_) : 0;
+    averageDocumentLength_ = nofDocuments_
+                                 ? (static_cast<float>(totalDocumentLength_) /
+                                    static_cast<float>(nofDocuments_))
+                                 : 0;
   }
 };
+
+void logWordNotFound(const string& word, size_t& wordNotFoundErrorMsgCount);
```

Comment:
This public function currently pollutes the global namespace, 
either you make it a `static` member function of the class where it's needed, 
or you only declare + define it statically in the `.cpp`file.